### PR TITLE
Unref timeouts to let process exit properly

### PR DESCRIPTION
**Pull request by johansteffner**

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -111,22 +111,22 @@ test('cache option', t => {
 
 test('promise support', async t => {
 	let i = 0;
-	const memoized = m(async () => i++);
+	const memoized = m(() => Promise.resolve(i++));
```
Comment (SamVerschueren) [inline test.js]: Why did you change this test file?

File: package.json [hunk 2]
```diff
@@ -32,7 +32,7 @@
 		"promise"
 	],
 	"dependencies": {
-		"map-age-cleaner": "^0.1.1",
+		"map-age-cleaner": "^0.1.3",
```
Comment (SamVerschueren) [inline package.json]: It should automatically be updated when reinstalling `mem` because of semver.

Comment (johansteffner) [reply package.json]: Yes, but when using tools like [renovate](https://renovatebot.com/), changes in a packages dependencies won't be detected.

File: test.js (hunk 1 above)
Comment (johansteffner) [reply test.js]: The xo linter complained about having an async function without an await inside. It's specified as `"xo": "*"` in dependencies, so i guess there was some change of this rule in some newer version.

File: package.json (hunk 2 above)
Comment (LinusU) [reply package.json]: @SamVerschueren So, when it actually fixes a bug, I think it's appropriate to release a new version with the minimum version bumped (with the `^` specifier) since the package isn't actually compatible with `0.1.1` and `0.1.2` in the sense that they are affected by the bug and thus are behaving buggy when consumed by someone else. Therefore I think that this should be merged and released as a patch release.

@johansteffner in the meantime you can bump just that package in your lock-file by running `npm update map-age-cleaner --depth=50` to just bump that package across your entire lock-file.

<details><summary>off topic:</summary>

I actually think that the only way to properly do automatic updates with Renovate and similar tools is to bump _the lock-file entry only_, whenever _any_ dependency is updated. There will be _a ton_ of bumps, but it's the only way to know exactly which dependency that broke your build.

If you only bump in you `package.json` file and let the package-lock regenerate itself you will consolidate many updated packages into only one PR + build; and you have only arbitrarily picked which packages you'll bump on...

</details>

Comment (johansteffner) [reply package.json]: @SamVerschueren do you agree with @LinusU reasoning about releasing this? Otherwise I'll close this PR.
